packages/thirdweb/src/utils/any-evm/zksync/isZkSyncChain.ts

```diff
@@ -1,12 +1,24 @@
+import { withCache } from "../../promise/withCache.js";
 import type { Chain } from "../../../chains/types.js";
 
-export function isZkSyncChain(chain: Chain) {
-  return (
-    chain.id === 324 ||
-    chain.id === 300 ||
-    chain.id === 302 ||
-    chain.id === 11124 ||
-    chain.id === 282 || // cronos zkevm testnet
-    chain.id === 388 // cronos zkevm mainnet
+export async function isZkSyncChain(chain: Chain) {
+  const stack = await getChainStack(chain.id);
+
+  return stack === "zksync-stack";
+}
+
+export async function getChainStack(chainId: number): Promise<string> {
+  return withCache(
+    async () => {
+      const res = await fetch(`https://${chainId}.rpc.thirdweb.com/stack`);
+      const data = await res.json();
+
+      if (res.ok) {
+        return data.stack;
+      } else {
+        return null;
```

Comment:
I can't remember the behavior of with cache, cam you double check that it won't cache null as a result?

Reply:
if you don't want cache simply throw on `!res.ok`

Reply:
yeah that's what i thought